adal/devApps/XFormsApp/SecondPage.cs

```diff
@@ -300,13 +331,44 @@ private async void AcquireTokenSilentWithBroker()
             {
                 Device.BeginInvokeOnMainThread(() =>
                 {
+                    this.testResult.Text = string.IsNullOrWhiteSpace(accessToken) ? "Result: Failure" : "Result: Success";
                     this.result.Text += "Result : " + output;
 
                     this.result.Text += "Logs : " + DrainLogs();
                 });
             }
         }
 
+        private void AcquireTokenWithBrokerButton_Clicked(object sender, EventArgs e)
+        {
+            switch (Device.RuntimePlatform)
+            {
+                case Device.iOS:
+                case Device.Android:
+                    AcquireTokenWithBroker();
+                    break;
+                default:
+                    this.result.Text = "UWP does not support broker. Use iOS or Android.";
```

Comment:
It would be safer to have case UWP - do stuff - default: throw. Although I would throw even in the UWP case - fail early and clearly ...